include/private/rdycoreimpl.h

```diff
@@ -234,7 +234,7 @@ PETSC_INTERN PetscErrorCode ReadCheckpointFile(RDy, const char *);
 
 // time series
 PETSC_INTERN PetscErrorCode InitTimeSeries(RDy);
-PETSC_INTERN PetscErrorCode AccumulateBoundaryFluxes(RDy, RDyBoundary boundary, PetscInt size, PetscInt ndof, PetscReal[size][ndof]);
+PETSC_INTERN PetscErrorCode AccumulateBoundaryFluxes(RDy, RDyBoundary boundary, PetscInt size, PetscInt ndof, PetscReal[size * ndof]);
```

Comment:
Actually, I think we can keep the VLA pointers--they aren't related to stack allocations. Does this produce different behavior?

Reply:
If I don't make the change, we will get a compiler error because the variable that is being passed to this function is a 1D pointer at https://github.com/RDycore/RDycore/pull/215/files#diff-6cfdc6b20c73c1aa99a7cbfcfda14bd9b0a765a72fa9942cda460313483e277fR17

Reply:
This hardly means anything in the declaration (in principle a compiler could check the invariant regarding allocation size, at least when `size` and `ndof` are compile-time constants, but I don't think I've seen that happen. I usually pass flat arrays through the public interface and dimension it with a VLA pointer within a function. If the build is happy (and we don't need to call the interface from C++), then either should be fine.

Reply:
It's your choice where to cast `(PetscReal (*)[ndof])flux`, either inside the function or at the call site. In general, I've found passing higher-dimensioned arrays often ends up brittle for C reasons so I usually pass flat arrays through public interfaces.

Reply:
I will change it to
```
PETSC_INTERN PetscErrorCode AccumulateBoundaryFluxes(RDy, RDyBoundary boundary, PetscInt size, PetscInt ndof, PetscReal*);
```